src/base/base-mixin.js

```diff
@@ -521,10 +523,45 @@ export class BaseMixin {
 
     generateSvg () {
         this._svg = this.root().append('svg');
+
+        this._svg.append('desc')
+            .attr('id', `desc-id-${this.__dcFlag__}`)
+            .html(`${this._svgDescription}`);
+
+        this._svg
+            .attr('tabindex', '0')
```

Comment:
Should this also be enabled/disabled by `keyboardAccessible`?

My concern is that it will probably change the behavior of existing applications.

Reply:
Sorry, my fault for not explaining the reasoning. I think it's useful to be able to turn the accessibility of internal chart elements on and off separately from the accessibility of the chart as a whole - depending on chart design, tabbing through many, many elements might not be desirable. `keyboardAccessible` is the method for the former and rolls together focusing and clicking. `svgDescription` is for the latter and covers focus and description. Since description is only useful on focused elements, I included `tabindex` by default.

The only scenario I can think of when this would result in an unexpected change is if the application changed `tabindex` to a value other than 0 which isn't [recommended](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex). I think as is, it's a "free" improvement for existing dashboards that mouse users shouldnt even notice.

Reply:
I think I get the design. What I meant is, what if someone has an existing dashboard with form elements, but the charts are not currently accessible using the tab key?

I think it's pretty common to have just a text entry field and a couple of selects, or whatever. I know I am lazy and I use the keyboard instead of the mouse in a familiar application, when I have to do something repetitive.

In other words, not all keyboard users are screen reader users.

With this change, the charts will be tabbable even though there isn't any interactivity or screen reader enabled for them. My concern is that it will make these simple form/dashboard applications harder to navigate, even though no functionality has been added.

Here's a very simple example: [complex reductions example](https://dc-js.github.io/dc.js/examples/complex-reduce.html). Before the change, just the links and the radio button group are accessible with the tab (on Chrome - on Firefox the links are not tabbable). After the change, the charts are tabbable but there isn't any functionality when they are focused.

Reply:
Thanks for taking the time to explain. I've made a few changes (and tests) so that we don't add any new tabbable elements unless the user explicitly tells us to by either setting `svgDescription` or `keyboardAccessible`. The latter will keep the class constructor name as default description, unless it's overwritten by `svgDescription`. The implementation feels a bit hacky, though.